### ISSUE TITLE
ISO standards on language in Polyglossia

### ISSUE THREAD

**Issue by pauloney** [reutenauer/polyglossia#59]
I would like to make a case here for the use of ISO standards in identifying language and scripts in Polyglossia, so we can all have an easier time in the future.

I can see that it (almost) uses ISO-15924 to name language scripts in the token "scripttag", but not quite. The three problems being:
- The ISO code tag requires the first letter to be a capital: Latn, Cyrl, Ethi, Arab, ...
- It is always a four letter-code, so the correct code for "lao" is "Laoo" and the correct code for N'ko is "Nkoo".
- Only 26 of the languages files contain the scripttag, I guess because all others use "Latn" but even so it would be nice to have it there so other programs that use and parse Polyglossis files can use it properly.

In the same vein some of the language files have a "langtag" which is almost ISO 639-3, but not exactly! The problems being:
- The code a lower-case 3-letter sequences, but inside Polyglossis is represented by an upper-case sequence.
- Some confusion on Turkish, on using "trk" or "tur". The correct code it "tur" and "trk"is not a code point on ISO-639-3.
  - Only 21 of the files have the language tag.

The ISO-639-3 could be used on many others instances as well, specially in the naming of files and in the user interface to load the languages. It would free us from aberrations like "portuges.ldf" or "gloss-portuges.ldf" which is inherited from the DOS era a the name of a language spelled incorrectly (missing the letter "u").  It would allow other programs that load/need Polyglossisa to start loading language and options using the ISO standards, but above all it would allow one to parse the files in a more efficient way.

Essentially, the problems that Polyglossia tackle are NOT problems of language, but problems of "locale". The northern African countries all use Arabic, but the name of the months vary widely among them, so in the case of the Arabic languages, the correct way would be to represent each file by a separate locale. We are a long ways from that, but implementing the scripttags and languagetags first would prepare the way to implement RFC-5646 later and correctly deal with that problems as well (http://tools.ietf.org/html/rfc5646).

Paulo Ney

**Comment by mb21** [comment]
+1

**Comment by mb21** [comment]
P.S. RFC-5646 is actually part of [BCP 47](https://tools.ietf.org/html/bcp47), so ultimately polyglossia should go for that, see #105

**Comment by jspitz** [comment]
Note that the `langtag` and `scripttag` is the OpenType tag.
https://docs.microsoft.com/en-us/typography/opentype/spec/languagetags
https://docs.microsoft.com/en-us/typography/opentype/spec/scripttags

Since these are passed to (and expected by) fontspec, we must use the OpenType standard here.

This is not to say that we shouldn't implement BCP-47 tags in parallel. But we have a separate request for this. See #226.
